From 3045b8c97e4b50ffedfe43ddde413c5af92e0a19 Mon Sep 17 00:00:00 2001
From: Yifei Teng <yifeit@google.com>
Date: Sat, 24 Jun 2023 07:25:55 +0000
Subject: [PATCH] Reland "[fuchsia] Re-enable -Wctad-maybe-unsupported"

This is a reland of commit 02f283f4aa8a3cfef60324969f9f20a4535ac79d

Original change's description:
> [fuchsia] Re-enable -Wctad-maybe-unsupported
>
> fxbug.dev/127411 has been fixed.
>
> Change-Id: If12f13bff93e652202d26f6f0c939bc212e77feb
> Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4641165
> Commit-Queue: Yifei Teng <yifeit@google.com>
> Reviewed-by: Patrick Thier <pthier@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#88573}

Change-Id: Ib391b3e61c0c368868102d3d0c9270a54e341d59
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4668071
Commit-Queue: Yifei Teng <yifeit@google.com>
Reviewed-by: Michael Achenbach <machenbach@chromium.org>
Cr-Commit-Position: refs/heads/main@{#88697}
---
 BUILD.gn | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index 94310ec7d04..0b9a109a6ff 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1440,13 +1440,10 @@ config("toolchain") {
       "-Wno-shadow",
     ]
 
-    # TODO(fuchsia:127411): Re-enable once FIDL bindings are compatible.
-    if (!is_fuchsia) {
-      # Google3 enables this warning, so we should also enable it to find issue
-      # earlier. See https://reviews.llvm.org/D56731 for details about this
-      # warning.
-      cflags += [ "-Wctad-maybe-unsupported" ]
-    }
+    # Google3 enables this warning, so we should also enable it to find issue
+    # earlier. See https://reviews.llvm.org/D56731 for details about this
+    # warning.
+    cflags += [ "-Wctad-maybe-unsupported" ]
 
     if (v8_current_cpu == "x64" || v8_current_cpu == "arm64" ||
         v8_current_cpu == "mips64el" || v8_current_cpu == "riscv64") {
-- 
2.35.1

